cpp/src/parquet/column_reader.cc

```diff
@@ -993,6 +996,9 @@ int64_t TypedColumnReaderImpl<DType>::ReadBatch(int64_t batch_size, int16_t* def
 
   *values_read = this->ReadValues(values_to_read, values);
   int64_t total_values = std::max(num_def_levels, *values_read);
+  if (total_values == 0) {
+    ParquetException::EofException("Read 0 values");
```

Comment:
is it possible to add in the number of additional values expected or is it not-knowable at this point?  Are there issues if somehow a data page/row group has different values?

Reply:
I would expect that we should see min(batch_size, num_buffered_values_ - num_decoded_values_) values. I will add this to the error message.

I don't entirely understand your second question. If a page has an inconsistent number of values from what is in the header, bad things can happen This PR is addressing one of those scenarios, where the reported number is too many causing an infinite loop. There are other scenarios however that I am not addressing. For example, if the reported number of values is lower than what actually exists, we can skip them.

Reply:
sorry for the second question I meant to ask if it has 0 values.  I think you answered this below.

Reply:
Ok, yes. I don't think the page size 0 was a problem(or legal) but I updated the code to check for the case.